tests/test_env.py

```diff
@@ -1,20 +1,86 @@
 from __future__ import annotations
 
-from pathlib import Path
+from typing import Dict
 
 import pytest
 
 
-@pytest.fixture()
-def example(testdir: pytest.Testdir) -> pytest.Testdir:
-    src = Path(__file__).parent / "example.py"
-    dest = Path(str(testdir.tmpdir / "test_example.py"))
-    dest.symlink_to(src)
-    return testdir
+@pytest.mark.parametrize("existing_env_vars,ini_contents,expected_env_vars", [
+    pytest.param(
+        {},  # empty dict means no preexisting environment variables
+        "[pytest]\nenv = MAGIC=alpha",
+        {"MAGIC": "alpha"},
+        id="When ini has a new key, Then it is added to env"
+    ),
+    pytest.param(
+        {"MAGIC": "alpha"},
+        "[pytest]\nenv = MAGIC=beta",
+        {"MAGIC": "beta"},
+        id="When ini has a key which already exists in env, Then the env value is overwritten"
+    ),
+    pytest.param(
+        {"MAGIC": "alpha"},
+        "[pytest]\nenv = D:MAGIC=beta",
+        {"MAGIC": "alpha"},
+        id="Given D flag, When ini has a key which already exists in env, Then the original env value is kept"
```

Comment:
bit too verbose for my taste, would prefer something like:

`D exist original kept`  so doesn't show up like an essay in the test report; same with all other tests